src/mcts/node.cc

```diff
@@ -237,20 +237,40 @@ uint64_t ReverseBitsInBytes(uint64_t v) {
 }
 }  // namespace
 
-V3TrainingData Node::GetV3TrainingData(GameResult game_result,
-                                       const PositionHistory& history) const {
+V3TrainingData Node::GetV3TrainingData(const PositionHistory& history,
+                                       bool checkmate) const {
   V3TrainingData result;
 
   // Set version.
   result.version = 3;
 
   // Populate probabilities.
-  float total_n = static_cast<float>(
-      GetN() - 1);  // First visit was expansion of it inself.
   std::memset(result.probabilities, 0, sizeof(result.probabilities));
-  for (const auto& child : Edges()) {
-    result.probabilities[child.edge()->GetMove().as_nn_index()] =
-        child.GetN() / total_n;
+  if (checkmate) {
```

Comment:
Why is that change needed?
If something less trivial than 5 lines change is needed to get nearly zero Elo improvement, we should not push that. That complicates the code with no benefit.

Reply:
This is exactly what Tilps was worried about earlier. The extensive comment describes why.

Reply:
And I wrote a comment a few up that I believe this patch could lead to training strength gains, even if in a same-weight comparison it doesn't add elo, it does affect training, and in a positive way IMO.